### PR TITLE
Storage API | normalize api landing page

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

create *Concepts and usage* section and move some previous sections that relates with concept as its sub section

create *Interfaces* section to include members in the api

reference to the spec https://storage.spec.whatwg.org/, also see the template https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/API_landing_page_template

### CODE REVIEW

File: files/en-us/web/api/storagemanager/persist/index.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ browser-compat: api.StorageManager.persist
 
 The **`persist()`** method of the {{domxref("StorageManager")}} interface requests permission to use persistent storage, and returns a {{jsxref('Promise')}} that resolves to `true` if permission is granted and bucket mode is persistent, and `false` otherwise.
 
+{{AvailableInWorkers}}
```
Comment (teoli2003) [inline files/en-us/web/api/storagemanager/persist/index.md:13]: _persist()_ is NOT available in workers.
Capture d’écran 2023-12-30 à 08 31 43

I think a note explicating it is needed here, as the interface is available in workers.

**Review by teoli2003 — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/storagemanager/persist/index.md (hunk 1 above)
Comment (skyclouds2001) [reply files/en-us/web/api/storagemanager/persist/index.md:13]: yes, my mistake

File: files/en-us/web/api/storagemanager/persist/index.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ browser-compat: api.StorageManager.persist
 
 The **`persist()`** method of the {{domxref("StorageManager")}} interface requests permission to use persistent storage, and returns a {{jsxref('Promise')}} that resolves to `true` if permission is granted and bucket mode is persistent, and `false` otherwise.
 
+> **Note:** This method is not available in [Web Workers](/en-US/docs/Web/API/Web_Workers_API), though the {{domxref("StorageManager")}} interface is available in [Web Workers](/en-US/docs/Web/API/Web_Workers_API).
```
Comment (teoli2003) [inline files/en-us/web/api/storagemanager/persist/index.md:13]:
```suggestion
> **Note:** This method is not available in [Web Workers](/en-US/docs/Web/API/Web_Workers_API), though the {{domxref("StorageManager")}} interface is.
```

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**